mmap.go

```diff
@@ -15,25 +13,8 @@ var (
 	ErrIndexOutOfBound = errors.New("offset out of mapped region")
 )
 
-// File provides an interface to a memory mapped file
-type File interface {
-	io.ReaderAt
-	io.WriterAt
-
-	ReadStringAt(dest *strings.Builder, offset int64) int
-	WriteStringAt(src string, offset int64) int
-	ReadUint64At(offset int64) uint64
-	WriteUint64At(num uint64, offset int64)
-
-	Lock() error
-	Unlock() error
-	Advise(advice int) error
-	Flush(flags int) error
-	Unmap() error
-}
-
-// FileImpl provides abstraction around a memory mapped file
-type FileImpl struct {
+// File provides abstraction around a memory mapped file
+type File struct {
```

Comment:
Agree with the changes, since the File is primarily used by bigqueue.

Why did you remove the io.ReaderAt and io.WriterAt embeds from the File struct?

Reply:
You can't have interfaces embedded into a struct. Only an interface can have embedded interface in it. The type `File` still satisfies `io.ReaderAt` and `io.WriterAt` and we still have test for that too here https://github.com/grandecola/mmap/blob/master/mmap_test.go#L550